src/averageAges.js

```diff
@@ -55,7 +65,16 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredChildren = people.filter(child =>
+    onlyWithSon ? people.some(mother =>
+      mother.name === child.mother) && child.sex === 'm'
+      : people.some(mother => mother.name === child.mother));
+
+  const ageDifference = filteredChildren.map(child =>
+    child.born - (people.find(mother => mother.name === child.mother)).born);
```

Comment:
Better create variable `motherBorn` for example, this way code would be easier to read.

Reply:
Thanks, made changes)